content/learning/operator/operator.md

```diff
@@ -65,10 +65,101 @@ Kubebuilder will generate the controller code only after generating CRDs, so eve
+
+3. Another problem that you can face relates to the creation and the update of resources in the cluster.
+Given that a controller must be _idempotent_ (i.e., a superfluous reconcile should have no effect and additional resources should be created or updated depending on whether already present in the cluster)), it is better to use [CreateOrUpdate](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.0/pkg/controller/controllerutil#CreateOrUpdate), a function that is intrinsically idempotent and it is part of the `controller-utils` package.
+To use the `CreateOrUpdate` function you have to first create an initial instance of the resource itself, specifying only its name and the namespace, if appropriate.
+Then, the function takes care of initializing the content of the resource with the current version retrieved from the cluster (if any), before executing the  _mutate function_ specified as a parameter.
+This has to configure the resource itself, e.g., adding labels, modifying annotations and describing specifications, to make it converge to the desired status.
+Finally, `CreateOrUpdate` takes care of saving the modified resource in the cluster.
+In other words, this function implements a very similar behavior to `kubectl apply`.
+
+    Here there is a CreateOrUpdate example for a deployment:
+    ```
+    deployment := corev1.Deployment{
+        ObjectMeta: metav1.ObjectMeta{
+            Name: "name", Namespace: "namespace"}
+            }
+    _, err = ctrl.CreateOrUpdate(ctx, r.Client, &deployment, func() error {
+            UpdateDeploymentSpec(&deployment)
+            return ctrl.SetControllerReference( &Owner, &deployment, r.Scheme)
+        })
+    ```
+
+
+4. While updating a resources, sometimes applying an Update() on a resource could trigger errors like: "_operation cannot be fulfilled on resource, the object has been modified_".
+This means that the controller is using a different version of the resource from the one present in the cluster; this of course leads to a failure when updating the resource.
+To overcome this problem you can use the function Patch() instead of Update().
+In fact, Patch() does not make any check on the resource version; a possible problem consists in the possibility to lose some updates (since the changes in a resource are compared thanks to the version of the resource itself).
+If the resource is watched only by one controller, you only may risk to overwrite your own changes and this may no matter thanks to idempotence. If the resource is owned by more than one controller, you should be very careful to use this function since you can loss some updates made by others controllers.
```

Comment:
```suggestion
If the resource is watched only by one controller, you only may risk to overwrite your own changes and this may not matter thanks to idempotence. If the resource is owned by more than one controller, you should be very careful to use this function since you can lose some updates made by others controllers.
```